roles/moving_objects_between_buckets/tasks/main.yml

```diff
@@ -0,0 +1,49 @@
+---
+- name: Run 'moving_object_between_buckets' role
+  module_defaults:
+    group/aws: "{{ aws_setup_credentials__output }}"
+
+  block:
+    # Checking required variables
+    - name: Fail when 'source_bucket' is undefined
+      ansible.builtin.fail:
+        msg: source bucket should be defined as source_bucket
+      when: source_bucket is undefined
+
+    - name: Fail when 'dest_bucket' is undefined
+      ansible.builtin.fail:
+        msg: destination bucket should be defined as dest_bucket
+      when: dest_bucket is undefined
+
+    # Obtaining specifc or all the source bucket objects
+    - name: Obtaining all the source bucket objects
+      amazon.aws.s3_object:
+        bucket: "{{ source_bucket }}"
+        mode: list
+        prefix: "{{ key_prefix | default('') }}"
+      register: source_bucket_info
+
+    # Transferring objects from source bucket to destination bucket
+    - name: Moving objects between source bucket to destination bucket
+      amazon.aws.s3_object:
+        bucket: "{{ dest_bucket }}"
+        object: "{{ item }}"
+        mode: copy
+        copy_src:
+            bucket: "{{ source_bucket }}"
+            object: "{{ item }}"
+      loop: "{{ source_bucket_info.s3_keys }}"
+
+    - name: Delete the source bucket objects
+      amazon.aws.s3_object:
+        bucket: "{{ source_bucket }}"
+        object: "{{ item }}"
+        mode: delobj 
+      loop: "{{ source_bucket_info.s3_keys }}"
+    
+    # Delete source s3 bucket when empty
+    - name: Delete source s3 bucket when empty
```

Comment:
probably let's add another task to verify that source bucket is empty before this task
and fail to delete if `bucket is not empty` or `delete_empty_source_bucket == False`